.github/workflows/test.yml

```diff
@@ -0,0 +1,46 @@
+name: Test
+on:
+  push:
+  pull_request:
+jobs:
+  test:
+    if: github.event_name != 'pull_request' || !contains('OWNER,MEMBER,COLLABORATOR', github.event.pull_request.author_association)
+    name: py${{ matrix.python }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: [3.7, 3.11]
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python }}
+    - run: pip install -U .[dev]
+    - run: pytest
+  deploy:
+    needs: [test]
+    name: PyPI Deploy
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - uses: actions/setup-python@v4
+      with:
+        python-version: '3.x'
+    - id: dist
+      uses: casperdcl/deploy-pypi@v2
+      with:
+        build: true
+        password: ${{ secrets.EQT_SECRET_TOKEN }}
+        upload: ${{ github.event_name == 'push' && startsWith(github.ref, 'refs/tags') }}
+    - if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
+      name: Release
+      run: |
+        changelog=$(git log --pretty='format:%d%n- %s%n%b---' $(git tag --sort=v:refname | tail -n2 | head -n1)..HEAD)
+        tag="${GITHUB_REF#refs/tags/}"
+        gh release create --title "Version $tag" --draft --notes "$changelog" "$tag" dist/${{ steps.dist.outputs.whl }} dist/${{ steps.dist.outputs.targz }}
```

Comment:
Does this automatically create the version description from the git logs?
I think we would prefer to use the change log we manually write in CHANGELOG.md

Reply:
yes but as per the updated `.github/workflows/README.md` instructions; it only creates a `--draft` release. Manual copyediting & approval of said draft is always required to make it public.

Reply:
> I like the automation

It's not about automation. Automation is merely a happy bonus side-effect. It's about DRY/SSoT (see [full explanation, including bugs I had to manually fix in the old release noted](https://github.com/paskino/qt-elements/pull/77#discussion_r1254839402)).

> I feel that the release text is related to a GitHub feature rather than an integral part of the software.

1. the GH release text was already being used before this PR
2. dev metadata (version history) is NOT meant to be an "integral part of software" (that's the whole point of git commits as well as commit messages)
3. best practice is to put release notes as comments in annotated tags if you want them in a place where a dev will find them easily sans GH web interface
4. release notes should link to relevant PRs (for context & discussion) which is distinctly inherently a GH-only (non-local-codebase) use case.


> Often I find myself editing the changelog differently than the commit messages. Maybe it is a good practice to write the commit messages better!

Still possible here. I do the same. The commit messages injected in the draft ensure:

1. I don't forget to mention something important
2. I'm embarrassed by poorly-written unfit-for-purpose messages, inspiring me to write better in future

> My real argument against deleting the `CHANGELOG.md` is that we tend to cluster changes on topic rather than commit.

1. that's what merge commits & tags are for, ergo what should be described in merge commit messages & annotated tags
2. the changelog was mirrored in two places, I'm simply proposing we delete one of those places

> As I understand it, there will still be the chance to edit the draft text, but how can we find the final text not on GitHub?

Annotated tags & (merge) commit messages.

Finally, overall harsh truth:

- high-level end users: only look at web material (PyPI, GH Readme, GH releases)
- devs: only look at git commits & git blame
- nobody: wants changelog file checked out in-repo

I really believe there is zero frustration if `changelog.txt` simply links to GH releases.

Reply:
added instructions & tests for annotated tags: 70c2eff